internal/packages/walk.go

```diff
@@ -432,27 +461,16 @@ func (r *symlinkResolver) follow(dir, base string) bool {
 		// Bazel to point to internal build directories.
 		return false
 	}
-	// See if the symlink points to a tree that has been already visited.
-	fullpath := filepath.Join(dir, base)
-	dest, err := filepath.EvalSymlinks(fullpath)
+	canonical, err := filepath.EvalSymlinks(filepath.Join(dir, base))
 	if err != nil {
 		return false
 	}
-	if !filepath.IsAbs(dest) {
```

Comment:
I think there's no need to call `Abs` here. Since `dir` is guaranteed to be an absolute path, `fullpath` and `dest` will also be absolute paths.